### PR TITLE
Add streaming parity tests, fixtures, and regression harness

### PR DESCRIPTION
**Pull request by cnkang**
## Summary by Sourcery

Add a comprehensive streaming parity and regression harness with corpus-driven, property-based, and boundary-focused tests for the Rust converter, plus supporting fixtures, utilities, fuzz targets, and guardrails.

Enhancements:
- Document additional guardrails for streaming tests, spec completion, warning triage, and Rust test infrastructure in AGENTS.md.
- Extend the Rust converter with utilities and helpers for charset handling, metadata docs, and streaming test support without changing core conversion behavior.
- Introduce a TOML-based registry of known streaming differences to keep fixture metadata and test expectations in sync.
- Add SonarCloud configuration to scope analysis, mark test directories, and suppress expected HTML lint warnings on corpus fixtures.

Build:
- Add new fuzz binaries for streaming no-panic and streaming chunk-split coverage and register them in the fuzz Cargo manifest and Makefile smoke target.

CI:
- Add a SonarCloud autoscan configuration file defining source/test roots and excluding intentional HTML corpus violations from specific web rules.

Documentation:
- Update AGENTS workflow guidance with new streaming guardrails, feature-gated hygiene rules, and evidence-generation expectations.
- Clarify Rust API examples in metadata docs to show explicit PageMetadata usage.

Tests:
- Add corpus-driven streaming parity tests comparing full-buffer, streaming, and incremental outputs across categories and large fixtures.
- Introduce targeted streaming regression suites for charset behavior, compressed inputs, high-risk structures (tables, nested lists, blockquotes, code), and failure paths (budget, timeout, malformed HTML, fallback).
- Add streaming chunk-boundary property tests and utilities to exercise single-byte, UTF-8 mid-character, tag-boundary, and fuzz-derived split patterns.
- Create shared streaming test support and comparison modules plus a TOML-backed known-differences registry to centralize expectations and evidence output.
- Generate new streaming-focused HTML fixtures and metadata (including large and high-risk cases) to drive parity and bounded-memory/TTFB evidence tests.

Chores:
- Apply minor Rust formatting and import ordering cleanups across streaming-related modules and FFI tests to satisfy style and lint guardrails.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive streaming conversion test suite with property-based and corpus-driven tests across chunk boundaries, failure paths, and high-risk structures.
  * Introduced new HTML test fixtures and metadata for streaming parity validation.
  * Enhanced fuzz testing with additional targets for streaming chunk-split and no-panic scenarios.

* **Chores**
  * Updated SonarQube Cloud configuration and project verification guidelines.

### CODE REVIEW

**Comment by cnkang** [comment]
@coderabbitai review

**Comment by cnkang** [comment]
@sourcery-ai review

**Comment by cnkang** [comment]
@greptile review

**Comment by cnkang** [comment]
@coderabbitai review

File: tests/streaming/known-differences.toml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+fixture_contains = "streaming/charset-mismatch.html"
+
+[[difference]]
+id = "DIFF-PARITY-EXPECTED-FALLBACK"
+description = "Fixtures with known table-heavy structures may fallback in streaming mode even when metadata is legacy"
+trigger = "streaming-fallback"
+reason = "Legacy corpus metadata does not yet mark all table-driven fallback fixtures"
+acceptable = true
+fix_version = "0.6.0"
+diff_contains = "streaming-fallback"
```
[the remark this answers is not in the document]
Comment (cnkang) [reply tests/streaming/known-differences.toml:163]: Addressed. I scoped the fallback suppressor so it no longer applies globally.

Changes made:
- DIFF-PARITY-EXPECTED-FALLBACK now has fixture_contains = "complex/".
- Added a separate scoped entry for simple/tables.html.

This prevents unrelated fixtures from being silently suppressed while preserving known legacy fallback cases. Included in commit c009e63.

File: components/rust-converter/tests/streaming_specific_regression.rs [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+                &run.markdown,
+                &known,
+            )
+            .unwrap_or_else(|err| panic!("{err}")),
+            Err(ConversionError::StreamingFallback { .. }) => {}
+            Err(err) => panic!(
+                "streaming conversion failed for {}: {err}",
+                fixture.display()
+            ),
+        }
```
[the remark this answers is not in the document]
Comment (cnkang) [reply components/rust-converter/tests/streaming_specific_regression.rs:210]: Addressed earlier. Full-buffer and streaming content-type are now aligned in this parity test.

Change:
- convert_full_buffer(..., Some("text/html; charset=UTF-8"), ...) in streaming_specific_regression.rs.

This was included in commit 4d6d6e6.

File: components/rust-converter/fuzz/fuzz_targets/streaming_no_panic.rs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    };
+
+    let mut conv = StreamingConverter::new(ConversionOptions::default(), MemoryBudget::default());
+    if set_content_type {
+        conv.set_content_type(Some("text/html; charset=UTF-8".to_string()));
+    }
+
+    let _ = conv.feed_chunk(payload);
+    let _ = conv.finalize();
+});
```
[the remark this answers is not in the document]
Comment (cnkang) [reply components/rust-converter/fuzz/fuzz_targets/streaming_no_panic.rs]: Addressed. Removed duplicate legacy no-panic fuzz target registration and file.

Changes:
- Removed streaming_no_panic bin entry from components/rust-converter/fuzz/Cargo.toml.
- Deleted components/rust-converter/fuzz/fuzz_targets/streaming_no_panic.rs.
- Kept canonical fuzz_streaming_no_panic target.

Included in commit c009e63.

File: components/rust-converter/fuzz/fuzz_targets/streaming_chunk_split.rs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    };
+
+    if single != split {
+        assert_eq!(
+            normalize_ascii_whitespace(&single),
+            normalize_ascii_whitespace(&split),
+            "streaming chunk split changed output beyond whitespace drift"
+        );
+    }
+});
```
[the remark this answers is not in the document]
Comment (cnkang) [reply components/rust-converter/fuzz/fuzz_targets/streaming_chunk_split.rs]: Addressed. Removed the legacy streaming_chunk_split fuzz target and standardized smoke invocation to canonical target names.

Changes:
- Removed streaming_chunk_split bin entry from components/rust-converter/fuzz/Cargo.toml.
- Deleted components/rust-converter/fuzz/fuzz_targets/streaming_chunk_split.rs.
- Updated Makefile fuzz smoke target to run fuzz_streaming_chunk_split.

Included in commit c009e63.

File: components/rust-converter/tests/streaming_high_risk_structures.rs [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+fn assert_fixture(name: &str) {
+    let path = fixture_path(name);
+    let fixture_name = format!("streaming/{name}.html");
+    let html = read_fixture(&path);
+    let meta = read_fixture_meta(&path);
+    let known = KnownDifferences::from_file(&known_differences_path())
+        .unwrap_or_else(|err| panic!("load known differences: {err}"));
+
+    let full = convert_full_buffer(&html, None, default_streaming_options())
+        .unwrap_or_else(|err| panic!("full-buffer conversion failed for {fixture_name}: {err}"));
```
[the remark this answers is not in the document]
Comment (cnkang) [reply components/rust-converter/tests/streaming_high_risk_structures.rs]: Addressed. Full-buffer and streaming calls are now content-type aligned in streaming_high_risk_structures.rs.

Change:
- convert_full_buffer(..., Some("text/html; charset=UTF-8"), ...).

Included in commit c009e63.

File: components/rust-converter/tests/streaming_parity.rs [hunk 6]
```diff
@@ -0,0 +1,511 @@
...
+fn make_html_of_size(target_bytes: usize) -> Vec<u8> {
+    let mut html = String::new();
+    html.push_str("<!DOCTYPE html><html><body>\n");
+    html.push_str("<p>bounded-memory-sentinel</p>\n");
+    while html.len() < target_bytes {
+        html.push_str("<!-- streaming-bounded-memory-padding -->\n");
+    }
+    html.push_str("</body></html>\n");
+    html.into_bytes()
+}
```
[the remark this answers is not in the document]
Comment (cnkang) [reply components/rust-converter/tests/streaming_parity.rs:396]: Addressed. The synthetic bounded-memory fixture now uses parser-visible padding instead of comments.

Change:
- Replaced  with <p>streaming-bounded-memory-padding</p> in make_html_of_size.

Verification:
- Ran cargo test --features streaming --test streaming_parity bounded_memory_and_ttfb_evidence_pack -- --ignored (pass).

Included in commit c009e63.

**Comment by cnkang** [comment]
@coderabbitai review
